src/components/HdIcon.vue

```diff
@@ -53,6 +67,27 @@ export default {
 
       return svg;
     },
+    addAccessibilityTags(svg) {
+      const randId = Math.floor(Math.random() * 1000);
+      const attrs = [];
+      if (this.title) {
+        const titleId = `${this.id || randId}-title`;
+        const titleTag = document.createElementNS('http://www.w3.org/2000/svg', 'title');
+        titleTag.innerHTML = this.title;
+        titleTag.id = titleId;
+        svg.insertBefore(titleTag, svg.firstChild);
+        attrs.push(titleId);
+      }
+      if (this.description) {
+        const descId = `${this.id || randId}-desc`;
+        const descTag = document.createElementNS('http://www.w3.org/2000/svg', 'desc');
+        descTag.id = descId;
+        svg.appendChild(descTag);
+        attrs.push(descId);
+      }
+      if (attrs.length) svg.setAttribute('aria-labelledby', attrs.join(' '));
+      svg.setAttribute('role', 'img');
```

Comment:
I don't think this is necessary, as it's always added (not covered by the conditions) so it will make more sense to add it to the template directly.

But, I'm personally in favor of not adding it at all, and leave this choice to the consumer of the component as we are binding all the passed attributes with `v-bind="$attrs"`, so if they want to have the role "img", they can just pass it as an attribute:
```vue
<HdIcon src="some-icon.svg" role="img" />
```